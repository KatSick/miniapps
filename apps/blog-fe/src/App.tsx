import { MyButton } from "@miniapps/design-system";
import { useState } from "react";
import viteLogo from "/vite.svg";
import reactLogo from "./assets/react.svg";

export const App: React.FC = () => {
  const [count, setCount] = useState(0);

  return (
    <div className="min-h-screen bg-gray-50 flex flex-col items-center justify-center p-8">
      <div className="flex gap-8 mb-8">
        <a href="https://vite.dev" target="_blank" rel="noopener">
          <img
            src={viteLogo}
            className="h-24 w-24 hover:drop-shadow-lg transition-all duration-300 hover:scale-110"
            alt="Vite logo"
          />
        </a>
        <a href="https://react.dev" target="_blank" rel="noopener">
          <img
            src={reactLogo}
            className="h-24 w-24 hover:drop-shadow-lg transition-all duration-300 hover:scale-110 animate-spin-slow"
            alt="React logo"
          />
        </a>
      </div>
      <h1 className="text-4xl font-bold text-gray-800 mb-8">Vite + React</h1>
      <div className="bg-white rounded-lg shadow-lg p-8 text-center">
        <MyButton onClick={() => setCount((count) => count + 1)}>count is {count}</MyButton>
        <p className="text-gray-600">
          Edit <code className="bg-gray-100 px-2 py-1 rounded text-sm font-mono">src/App.tsx</code>{" "}
          and save to test HMR
        </p>
      </div>
      <p className="text-gray-500 text-sm mt-8 text-center max-w-md">
        Click on the Vite and React logos to learn more
      </p>
    </div>
  );
};
