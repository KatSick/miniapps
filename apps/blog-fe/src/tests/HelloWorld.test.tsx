import { expect, test, vi } from "vitest";
import { render } from "vitest-browser-react";
import { App } from "@/App";

test("renders name", async () => {
  const { getByText } = render(<App />);
  const element = await vi.waitFor(() => getByText("Vite + React"));
  expect(element).toMatchSnapshot();
});
