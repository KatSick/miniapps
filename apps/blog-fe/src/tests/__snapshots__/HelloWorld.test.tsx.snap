// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders name 1`] = `"body >> internal:text="Vite + React"i"`;
